pkg/apiserver/auditing/types.go

```diff
@@ -37,16 +41,18 @@ type Auditing interface {
 }
 
 type auditing struct {
-	lister  v1alpha1.WebhookLister
-	cache   chan *auditv1alpha1.EventList
-	backend *Backend
+	lister         v1alpha1.WebhookLister
+	resourceGetter *resourcesv1alpha3.ResourceGetter
+	cache          chan *auditv1alpha1.EventList
+	backend        *Backend
 }
 
-func NewAuditing(lister v1alpha1.WebhookLister, url string, stopCh <-chan struct{}) Auditing {
+func NewAuditing(informers informers.InformerFactory, url string, stopCh <-chan struct{}) Auditing {
 
 	a := &auditing{
-		lister: lister,
-		cache:  make(chan *auditv1alpha1.EventList, DefaultCacheCapacity),
+		lister:         informers.KubeSphereSharedInformerFactory().Auditing().V1alpha1().Webhooks().Lister(),
+		resourceGetter: resourcesv1alpha3.NewResourceGetter(informers),
```

Comment:
It's not necessary to create a resourceGetter since just devops needed, we'd better create devopsGetter only